hive-event-listeners/apiary-gluesync-listener/src/main/java/com/expediagroup/apiary/extensions/gluesync/listener/service/GlueTableService.java

```diff
@@ -0,0 +1,122 @@
+
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.BatchCreatePartitionRequest;
+import com.amazonaws.services.glue.model.CreateTableRequest;
+import com.amazonaws.services.glue.model.DeleteTableRequest;
+import com.amazonaws.services.glue.model.GetPartitionsRequest;
+import com.amazonaws.services.glue.model.Partition;
+import com.amazonaws.services.glue.model.PartitionInput;
+import com.amazonaws.services.glue.model.UpdateTableRequest;
+
+public class GlueTableService {
+  private static final Logger log = LoggerFactory.getLogger(GlueTableService.class);
+  public static final String APIARY_GLUESYNC_SKIP_ARCHIVE_TABLE_PARAM = "apiary.gluesync.skipArchive";
+
+
+  private final AWSGlue glueClient;
+  private final HiveToGlueTransformer transformer;
+
+  public GlueTableService(AWSGlue glueClient, String gluePrefix) {
+    this.glueClient = glueClient;
+    this.transformer = new HiveToGlueTransformer(gluePrefix);
+    log.debug("ApiaryGlueSync created");
+  }
+
+  public void create(Table table) {
+    CreateTableRequest createTableRequest = new CreateTableRequest()
+        .withTableInput(transformer.transformTable(table))
+        .withDatabaseName(transformer.glueDbName(table));
+    glueClient.createTable(createTableRequest);
+    log.info(table + " table created in glue catalog");
+  }
+
+  public void update(Table table) {
+    boolean skipArchive = shouldSkipArchive(table);
+
+    UpdateTableRequest updateTableRequest = new UpdateTableRequest()
+        .withSkipArchive(skipArchive)
+        .withTableInput(transformer.transformTable(table))
+        .withDatabaseName(transformer.glueDbName(table));
+    glueClient.updateTable(updateTableRequest);
+    log.info(table + " table updated in glue catalog");
+  }
+
+  public void delete(Table table) {
+    DeleteTableRequest deleteTableRequest = new DeleteTableRequest()
+        .withName(table.getTableName())
+        .withDatabaseName(transformer.glueDbName(table));
+    glueClient.deleteTable(deleteTableRequest);
+    log.info(table + " table deleted from glue catalog");
+  }
+
+  /**
+   * Function to copy partitions using BatchCreatePartitionRequest but knowing the partition input list limit is 100,
+   * however input parameter can have larger list
+    */
+  public void copyPartitions(Table table, List<Partition> partitions) {
+    List<PartitionInput> partitionInputs = partitions.stream()
+        .map(this::convertToPartitionInput)
+        .collect(Collectors.toList());
+    int partitionCount = partitionInputs.size();
+    int batchSize = 100;
+    for (int i = 0; i < partitionCount; i += batchSize) {
+      int end = Math.min(i + batchSize, partitionCount);
+      List<PartitionInput> batch = partitionInputs.subList(i, end);
+      BatchCreatePartitionRequest batchCreatePartitionRequest = new BatchCreatePartitionRequest()
```

Comment:
The batch renames are sequential, right?

Reply:
Yes

Reply:
Can this be optimised through parallelisation if needed?